i18next-parser.config.js

```diff
@@ -0,0 +1,116 @@
+import { locales } from "./smartling.config.js";
```

Comment:
This file is configuration for the tool that scans the repo for strings to translate. It's mostly using the defaults. 

The important settings we changed are `keySeparator`, and `namespaceSeparator` for the reasons mentioned in the comments.

The list of locales is imported from the smartling config file. 

`compatibilityJSON` is used to give us a simpler syntax for plurals (we may want to figure out and use the newer syntax at some point, but I wasn't clear if smartling/po files supported it). 

`input` specifies that all ts/tsx files in `src` should be scanned.